### PR TITLE
Change Ubuntu RID to Linux

### PR DESCRIPTION
**Pull request by khyperia**
Also remove osx RID version

Information on RIDs: https://docs.microsoft.com/en-us/dotnet/core/rid-catalog

(in particular, validation for the osx RID version removal: "`osx-x64` (.NET Core 2.0 or later versions, minimum version is `osx.10.12-x64`)")

### CODE REVIEW

File: build/Targets/Settings.props [hunk 1]
```diff
@@ -158,7 +158,7 @@
   <!-- Unix specific settings -->
   <PropertyGroup Condition="'$(OS)' != 'Windows_NT'">
     <RoslynDebugType>portable</RoslynDebugType>
-    <RoslynPortableRuntimeIdentifiers>ubuntu.14.04-x64;ubuntu.16.04-x64;osx.10.12-x64</RoslynPortableRuntimeIdentifiers>
+    <RoslynPortableRuntimeIdentifiers>linux-x64;osx-x64</RoslynPortableRuntimeIdentifiers>
```
Comment (jaredpar) [inline build/Targets/Settings.props]: Do we need to keep this is as -x64 still? Don't want to over complicate this change but curiosu if we should explore removing that too.

File: build/scripts/tests.sh [hunk 2]
```diff
@@ -15,11 +15,20 @@ binaries_path="${root_path}"/Binaries
...
 
+UNAME="$(uname)"
+if [ "$UNAME" == "Darwin" ]; then
+    runtime_id=osx-x64
+elif [ "$UNAME" == "Linux" ]; then
+    runtime_id=linux-x64
+else
+    echo "Unknown OS: $UNAME" 1>&2
+    exit 1
+fi
```
Comment (jaredpar) [inline build/scripts/tests.sh]: This is duplicated between tests.sh and cibuild.sh. Could we add a helper to build-util.sh and share it?

File: src/Deployment/DeployToolsetCompiler/project.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   },
   "runtimes": {
     "win7": { },
-    "ubuntu.14.04": { },
-    "osx.10.10": { }
+    "linux": { },
+    "osx": { }
```
Comment (jaredpar) [inline src/Deployment/DeployToolsetCompiler/project.json]: Wait ... we still have a project.json file???? I will delete this soon.

File: build/scripts/tests.sh (hunk 2 above)
Comment (khyperia) [reply build/scripts/tests.sh]: This PR deletes the dynamic RID detection in cibuild.sh https://github.com/dotnet/roslyn/pull/22807 so I don't think it's worth it to factor it out.

File: build/Targets/Settings.props (hunk 1 above)
Comment (khyperia) [reply build/Targets/Settings.props]: No idea... I'll run a local build and see if it explodes, and see how bad it is 🙂

File: build/scripts/tests.sh (hunk 2 above)
Comment (khyperia) [reply build/scripts/tests.sh]: (that PR just merged, so I've rebased this and got rid of this entirely)

**Comment by khyperia** [comment]
Nope on the removal of x64:

`error : Project is targeting runtime 'linux' but did not resolve any runtime-specific packages for the 'Microsoft.NETCore.App' package.  This runtime may not be supported by .NET Core.`

**Review by agocke — APPROVED** [review]
LGTM

File: src/Deployment/DeployToolsetCompiler/project.json (hunk 3 above)
Comment (benmvp) [reply src/Deployment/DeployToolsetCompiler/project.json:9]: Fixed: added control-flow comments around programmatic execution, recovery finalization, and model-step persistence.

**Pull request merged**